### PR TITLE
ESQL: Remove qualifier from attrs

### PR DESCRIPTION
**Pull request by alex-spies**
In ES|QL's analyzer, we resolve names to field attributes, like `some_field` in `FROM some_idx | KEEP some_field`.

The `Attribute` type, being copied from (S)QL, has a name and a qualified name, e.g. `some_field` and `some_idx.some_field`.

**However**, we never use the qualifier (`some_idx` in above's example). In the analyzer, the qualifier is always set to `null` and during optimization, we often do not take the qualified name into account, because we know the name alone matters. 

This is due to ES|QL's treatment of names, which is different from SQL's: at each point in the query, the (intermediate) result's columns must have unique names.

Let's remove `Attribute.qualifier`. As it is right now, it is confusing because one doesn't know if it may or may not contain relevant info, and especially the `Attribute.cluster` private member looks like it does something for cross cluster queries, but it really doesn't.

### CODE REVIEW

**Review by luigidellaquila — APPROVED** [review]
Thanks Alex, LGTM

> In the analyzer, the qualifier is always set to null

This alone justifies the removal IMHO.

There is a chance that we'll need similar concepts when we implement joins, but my gut feeling says that we'll adopt completely different solutions (as we do for LOOKUP and ENRICH). 

And in general, I'd rather add a piece of information only when I need it, to avoid biases in the design of future features.

**Review by astefan — COMMENTED** [review]
For the moment, let's keep it as is.

**Review by astefan — APPROVED** [review]
LGTM.
But I'd like @bpintea and @costin to weigh in, as well.
For the record, I am ok removing this for now (https://github.com/elastic/elasticsearch/issues/101177 might use it) because it's code that's not used. It was a valid code in ES SQL especially, but ES|QL for now doesn't use it and it creates confusion among contributors.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java [hunk 1]
```diff
@@ -823,8 +814,8 @@ private DataType[] allowedEnrichTypes(String matchType) {
     }
 
     private static List<Attribute> resolveAgainstList(UnresolvedNamePattern up, Collection<Attribute> attrList) {
-        UnresolvedAttribute ua = new UnresolvedAttribute(up.source(), up.pattern(), null);
-        Predicate<Attribute> matcher = a -> up.match(a.name()) || up.match(a.qualifiedName());
+        UnresolvedAttribute ua = new UnresolvedAttribute(up.source(), up.pattern());
+        Predicate<Attribute> matcher = a -> up.match(a.name()) || up.match(a.name());
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java:818]:
```suggestion
        Predicate<Attribute> matcher = a -> up.match(a.name());
```

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/AnalyzerRules.java [hunk 2]
```diff
@@ -76,18 +76,10 @@ public static List<Attribute> maybeResolveAgainstList(
...
-        final boolean qualified = u.qualifier() != null;
 
         Predicate<Attribute> predicate = a -> {
-            return qualified ? Objects.equals(qualifier, a.qualifiedName()) :
-            // if the field is unqualified
-            // first check the names directly
-                (Objects.equals(name, a.name()))
-                    // but also if the qualifier might not be quoted and if there's any ambiguity with nested fields
-                    || Objects.equals(name, a.qualifiedName());
+            return Objects.equals(name, a.name());
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/AnalyzerRules.java:82]: nit: lambda can be simplified.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/AnalyzerRules.java [hunk 3]
```diff
@@ -119,32 +111,20 @@ public static List<Attribute> maybeResolveAgainstList(
...
+        }).map(a -> "line " + a.sourceLocation().toString().substring(1) + " [" + a.name() + "]").toList();
 
         return singletonList(
             ua.withUnresolvedMessage(
-                "Reference ["
-                    + ua.qualifiedName()
-                    + "] is ambiguous (to disambiguate use quotes or qualifiers); "
-                    + "matches any of "
-                    + refs
+                "Reference [" + ua.name() + "] is ambiguous (to disambiguate use quotes or qualifiers); " + "matches any of " + refs
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/AnalyzerRules.java:127]: The message here should have the qualifiers suggestion removed, right?

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Alias.java [hunk 4]
```diff
@@ -32,6 +32,7 @@ public final class Alias extends NamedExpression {
     public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(NamedExpression.class, "Alias", Alias::new);
 
     private final Expression child;
+    // TODO: unused, remove
```
Comment (bpintea) [inline x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Alias.java:35]: Why not do it now? Along with `qualifiedName()` and `qualifier()`?

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java [hunk 5]
```diff
@@ -152,7 +152,7 @@ else if (plan instanceof Project project) {
                         // otherwise point to it
                         else {
                             // since avoids creating field copies
-                            projection = new Alias(f.source(), f.name(), f.qualifier(), nullAlias.toAttribute(), f.id());
+                            projection = new Alias(f.source(), f.name(), null, nullAlias.toAttribute(), f.id());
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java:155]: Is there no c'tor excluding the qualifier?

**Review by bpintea — APPROVED** [review]
Lgtm

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Alias.java (hunk 4 above)
Comment (alex-spies) [reply x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Alias.java:35]: I guess I wanted to keep the blast radius smaller, but it turns out it's not so big. Removing!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java (hunk 5 above)
Comment (alex-spies) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java:155]: Added one!

**Comment by alex-spies** [comment]
Thanks for the reviews @luigidellaquila , @astefan and @bpintea !

**Pull request merged**